iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,343 @@
+
+    const auto successfullySent = m_runtime->sendRequestToRouDi(m_sendBuffer, m_receiveBuffer);
+
+    EXPECT_EQ(false, successfullySent);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSenderIsSuccessful)
+{
+    const auto senderPort = m_runtime->getMiddlewareSender(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), senderPort->m_serviceDescription);
+    EXPECT_EQ(22u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_DefaultArgs)
+{
+    const auto senderPort = m_runtime->getMiddlewareSender(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, senderPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareSender_SenderlistOverflow)
+{
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER; ++i)
+    {
+        m_runtime->getMiddlewareSender(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiverIsSuccessful)
+{
+    auto receiverPort = m_runtime->getMiddlewareReceiver(
+        iox::capro::ServiceDescription(99u, 1u, 20u), m_runnableName, iox::runtime::PortConfigInfo(11u, 22u, 33u));
+
+    EXPECT_EQ(iox::capro::ServiceDescription(99u, 1u, 20u), receiverPort->m_serviceDescription);
+    EXPECT_EQ(22u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(33u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_DefaultArgs)
+{
+    auto receiverPort = m_runtime->getMiddlewareReceiver(iox::capro::ServiceDescription(99u, 1u, 20u));
+
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.deviceId);
+    EXPECT_EQ(0u, receiverPort->m_memoryInfo.memoryType);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareReceiver_ReceiverlistOverflow)
+{
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0u; i < iox::MAX_PORT_NUMBER + 1; ++i)
+    {
+        m_runtime->getMiddlewareReceiver(iox::capro::ServiceDescription(i, i + 1u, i + 2u));
+    }
+
+    EXPECT_TRUE(errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetServiceRegistryChangeCounter_OfferStopOfferService)
+{
+    auto initialValue = m_runtime->getServiceRegistryChangeCounter();
+    auto cout = initialValue->load();
```

Comment:
strange variable name, meant count? Why not initialCount?

Reply:
return value of getServiceRegistryChangeCounter() is atomic and updates when there is change in service, so i renamed initialValue to serviceCounter, and count to initialCount